Packs/Druva/Integrations/DruvaEventCollector/DruvaEventCollector.py

```diff
@@ -0,0 +1,289 @@
+from urllib.parse import quote
+
+import demistomock as demisto
+from CommonServerPython import *
+import urllib3
+import base64
+
+MIN_FETCH = 1
+MAX_FETCH = 10000
+MAX_EVENTS_API_CALL = 500  # As a limitation of the API, we can only retrieve 500 events at a time
+# Disable insecure warnings
+urllib3.disable_warnings()
+
+""" CONSTANTS """
+
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%SZ"
+DATE_FORMAT_FOR_TOKEN = "%m/%d/%Y, %H:%M:%S"
+VENDOR = "Druva"
+PRODUCT = "Druva"
+
+""" CLIENT CLASS """
+
+
+class Client(BaseClient):
+
+    def __init__(self, base_url: str, client_id: str, secret_key: str, max_fetch: int, verify: bool, proxy: bool):
+
+        super().__init__(base_url=base_url, verify=verify, proxy=proxy)
+        self.credentials = f"{client_id}:{secret_key}"
+        self.max_fetch = max_fetch
+        self._max_fetch_validation()
+        self.login()
+
+    def login(self):
+        """
+        In this method, the validity of the Access Token is checked, since the Access Token has a 30 minutes validity period.
+        Refreshes the token as needed.
+        """
+        now = datetime.utcnow()
+
+        if (cache := get_integration_context()) and (token := cache.get("Token")):
+            expiration_time = datetime.strptime(
+                cache["expiration_time"], DATE_FORMAT_FOR_TOKEN
+            )
+
+            # check if token is still valid, and use the old one. otherwise regenerate a new one
+            if (expiration_time - now).total_seconds() > 0:
+                demisto.debug("No need to regenerate the token, it is still valid")
```

Comment:
```suggestion
            if (seconds_left := (expiration_time - now).total_seconds()) > 0:
                demisto.debug("No need to regenerate the token, it is still valid for {seconds_left} more seconds")
```
not necessary, but would delight a developer debugging it in the future